### PR TITLE
[RNMobile] Split Para/Heading blocks on enter.KEY - step 1

### PR DESCRIPTION
**Pull request by daniloercoli**
This PR does add a very simple implementation of `onHTMLContentWithCursor` method to RichText. The code doesn't check much at the moment, but just call `onSplit` on the outer Block, if it has `onSplit` defined.

`onSplit`on both Para and Heading blocks, is very minimal, it just calls `insertAfter` without checking the position of the cursor, or any other condition.

### CODE REVIEW

File: packages/block-library/src/heading/edit.native.js [hunk 1]
```diff
@@ -24,6 +24,25 @@ import './editor.scss';
 const minHeight = 50;
 
 class HeadingEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
```
Comment (mzorz) [inline packages/block-library/src/heading/edit.native.js]: nitpick: can we remove this extra line here?

File: packages/editor/src/components/rich-text/index.native.js [hunk 2]
```diff
@@ -49,8 +49,22 @@ export class RichText extends Component {
...
+		if ( ! this.props.onSplit ) {
+			// insert the \n char instead?
+			return;
+		}
+		this.splitContent( htmlText, start, end );
+	}
+
+	splitContent( htmlText, start, end ) {
+		const { onSplit } = this.props;
+
```
Comment (mzorz) [inline packages/editor/src/components/rich-text/index.native.js]: nitpick #2: can we remove this extra line here?

File: packages/block-library/src/paragraph/edit.native.js [hunk 3]
```diff
@@ -8,12 +8,31 @@ import { View } from 'react-native';
...
-import { parse } from '@wordpress/blocks';
+import { parse, createBlock } from '@wordpress/blocks';
 import { RichText } from '@wordpress/editor';
 
 const minHeight = 50;
 
 class ParagraphEdit extends Component {
+	constructor() {
+		super( ...arguments );
+
```
Comment (mzorz) [inline packages/block-library/src/paragraph/edit.native.js]: nitpick #3: let's remove the extra line here

File: packages/editor/src/components/rich-text/index.native.js [hunk 4]
```diff
@@ -49,8 +49,22 @@ export class RichText extends Component {
 	}
 
 	// eslint-disable-next-line no-unused-vars
-	onHTMLContentWithCursor( htmlText, cursorPosition ) {
-		// Descriptive placeholder: This logic still needs to be implemented.
+	onHTMLContentWithCursor( htmlText, start, end ) {
+		if ( ! this.props.onSplit ) {
+			// insert the \n char instead?
```
Comment (mzorz) [inline packages/editor/src/components/rich-text/index.native.js]: Let's add a ToDo comment here to make sure this is something we come back to at a follow up stage

**Comment by daniloercoli** [comment]
Ready for another round @mzorz! 🙇

File: packages/block-library/src/heading/edit.native.js [hunk 5]
```diff
@@ -24,6 +24,24 @@ import './editor.scss';
 const minHeight = 50;
 
 class HeadingEdit extends Component {
+	constructor() {
+		super( ...arguments );
+		this.splitBlock = this.splitBlock.bind( this );
+	}
+
+	// eslint-disable-next-line no-unused-vars
+	splitBlock( htmlText, start, end ) {
```
Comment (hypest) [inline packages/block-library/src/heading/edit.native.js]: @daniloercoli , can we perhaps go closer to the interface of the web-side counterpart, wdyt? https://github.com/WordPress/gutenberg/blob/master/packages/block-library/src/heading/edit.js#L52.

Comment (daniloercoli) [reply packages/block-library/src/heading/edit.native.js]: Oh yes, we will probably go toward that direction in the short future. For now i'd like to go with the current implementation, and get the ball rolling.

**Review by hypest — APPROVED** [review]
LGTM!

**Review by mzorz — APPROVED** [review]
Thanks for addressing the small-ish comments! LGTM :shipit:

**Pull request merged**